geopandas/sindex.py

```diff
@@ -1,25 +1,235 @@
-from geopandas import base
+from warnings import warn
+from collections import namedtuple
 
-if base.HAS_SINDEX:
-    from rtree.index import Index as RTreeIndex
+from pandas import notnull
+import numpy as np
 
+from . import _compat as compat
 
-class SpatialIndex(RTreeIndex):
```

Comment:
we might need to keep this class for backwards compatibility fo now (but it could just be an alias of either PyGEOSSTRTreeIndex or RtreeIndex, depending on which one is used?)

Reply:
The `__init__` arguments would be different, so I don't think it would be as simple as setting `SpatialIndex = RTreeIndex`. That said, the original `SpatialIndex` is pretty small, I have no problem just leaving it there if you think that's best.

Reply:
> The __init__ arguments would be different, so I don't think it would be as simple as setting SpatialIndex = RTreeIndex.

That's what I meant with my other comment: an option is also to keep some compatibility in the `__init__`, so by ensuring we accept both the stream that the "old" SpatialIndex accepted, and the geometry series as the new version does.

Reply:
Oh I see what you mean. I think we should pick if we either (1) keep the old SpatialIndex class around for back compatibility for anyone who was importing and using it directly or (2) don't keep the old SpatialIndex but make RTreeIndex accept the arguments from the old spatial index.

Reply:
Option 1 is probably fine for now. We can think about a "unified" interface later